### PR TITLE
Finish applying Clippy suggestions.

### PR DESCRIPTION
**Pull request by sstangl**
This finishes applying Clippy's basic code cleanup suggestions. It turns out that there were not many!

I've only applied Clippy suggestions that should be obviously-unobjectionable in review, based off discussion from the previous patch.

### CODE REVIEW

File: core/lib/src/router/route.rs [hunk 1]
```diff
@@ -77,17 +77,16 @@ impl Route {
...
     /// ```
-    pub fn new<S>(m: Method, path: S, handler: Handler) -> Route
+    pub fn new<S>(method: Method, path: S, handler: Handler) -> Route
         where S: AsRef<str>
     {
         let uri = Uri::from(path.as_ref().to_string());
+        let rank = default_rank(&uri);
+
         Route {
+            method, handler, uri, rank,
```
Comment (jebrosen) [inline core/lib/src/router/route.rs]: This line (and again in `ranked`) now contains more than one of the fields, and they are in a different order. I'm not entirely sold on that.

Comment (sstangl) [reply core/lib/src/router/route.rs]: Sergio asked me to do that on the last patch, so I applied it again here.

File: core/codegen/src/decorators/route.rs [hunk 2]
```diff
@@ -87,12 +87,10 @@ impl RouteParams {
     }
 
     fn generate_data_statement(&self, ecx: &ExtCtxt) -> Option<Stmt> {
-        let param = self.data_param.as_ref().map(|p| &p.value);
-        let arg = param.and_then(|p| self.annotated_fn.find_input(&p.node.name));
-        if param.is_none() {
-            return None;
-        } else if arg.is_none() {
-            self.missing_declared_err(ecx, param.unwrap());
+        let param = self.data_param.as_ref().map(|p| &p.value)?;
```
Comment (SergioBenitez) [inline core/codegen/src/decorators/route.rs]: Clippy was able to find this change? That's awesome.

File: core/lib/src/config/custom_values.rs [hunk 3]
```diff
@@ -211,7 +211,7 @@ pub fn u64(conf: &Config, name: &str, value: &Value) -> Result<u64> {
 
 pub fn u16(conf: &Config, name: &str, value: &Value) -> Result<u16> {
     match value.as_integer() {
-        Some(x) if x >= 0 && x <= (u16::max_value() as i64) => Ok(x as u16),
+        Some(x) if x >= 0 && x <= (i64::from(u16::max_value())) => Ok(x as u16),
```
Comment (SergioBenitez) [inline core/lib/src/config/custom_values.rs]: Is there a reason to prefer this over the `as` coercion? The former seems clearer to me.

Comment (svmnotn) [reply core/lib/src/config/custom_values.rs]: I had the same question, this seems to be the answer as to why clippy does this lint:
> Rust’s `as` keyword will perform many kinds of conversions, including silently lossy conversions. Conversion functions such as `i32::from` will only perform lossless conversions. Using the conversion functions prevents conversions from turning into silent lossy conversions if the types of the input expressions ever change, and make it easier for people reading the code to know that the conversion is lossless.

File: core/lib/src/router/route.rs (hunk 1 above)
Comment (SergioBenitez) [reply core/lib/src/router/route.rs]: Yeah. This one seems messier, though. Oh well; let's just do this for consistency.

**Comment by SergioBenitez** [comment]
@sstangl Would you be able to rebase?

**Comment by SergioBenitez** [comment]
Awesome! Kept the vast majority of this (minus the `X::from` instead of `as`) and merged in 9bf585496. Thanks again, @sstangl! :)

**Pull request closed without merging**